### PR TITLE
Redfish: Implements Redfish Assembly Health

### PR DESCRIPTION
**Pull request by ChicagoDuan**
Look for the OperationalStatus to set the Assembly Health.
If the OperationalStatus is false, then set the Health to "Critical"
If the OperationalStatus is true, then the Health is "OK"

Signed-off-by: Chicago Duan <duanzhijia01@inspur.com>

### CODE REVIEW

**Comment by ChicagoDuan** [comment]
Hi @SunnySrivastava1984  @jinuthomas, I don't have assemblies. Can you help me test this commit? 
Thank you!

File: redfish-core/lib/assembly.hpp [hunk 1]
```diff
@@ -195,6 +195,50 @@ inline void
...
+                                        return;
+                                    }
+
+                                    nlohmann::json& assemblyArray =
+                                        aResp->res.jsonValue["Assemblies"];
+                                    nlohmann::json& assemblyData =
+                                        assemblyArray.at(assemblyIndex);
+
+                                    // Set the default Health to OK
+                                    assemblyData["Status"]["Health"] = "OK";
```
Comment (gtmills) [inline redfish-core/lib/assembly.hpp:219]: Do we want a "Status""State" ?
Should we move this outside of this else if so we could have a health of OK if this interface doesn't exist? 
Maybe set 
"Status""State" = "Enabled" 
"Status""Health" = "OK"
outside of this else if?

Comment (ChicagoDuan) [reply redfish-core/lib/assembly.hpp]: Done

**Comment by gtmills** [comment]
jenkins run test please

**Comment by gtmills** [comment]
@ChicagoDuan Can this get pushed upstream?

**Review by gtmills — APPROVED** [review]

**Comment by ChicagoDuan** [comment]
> @ChicagoDuan Can this get pushed upstream?

Hi @gtmills , I pushed it to the gerrit :  https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/50317, but CI failed. It seems that the parent commit needs rebase.  

Hi @SunnySrivastava1984 , Can you rebase this commit : Inventory properties via Assembly schema on bmcweb https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/39574/41 . Thanks!

**Pull request merged**